OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDetailFragment.java

```diff
@@ -303,6 +303,10 @@ private void configureEnrollButton() {
 
         if (mEnrolled) {
             mEnrollButton.setText(R.string.view_course_button_text);
+        } else if (courseDetail.invitation_only) {
```

Comment:
Until and unless we start receiving the `invitation_only` flag from server everytime, this check will always make the app throw NPE.